geo-replication/syncdaemon/primary.py

```diff
@@ -1043,7 +1052,39 @@ def process_change(self, change, done, retry):
                ec[self.POS_TYPE] not in ["UNLINK", "RMDIR", "MKNOD"]:
                 continue
 
-            if et == self.TYPE_ENTRY:
+            # Passive syncs the MKNOD operation alone.
+            if is_passive and et == self.TYPE_ENTRY:
```

Comment:
This code causes a crash while parsing D lines since (IndexError: list index out of range).
I"ll push a fix once I understand If a D entry suppuse to be ignored by a passive worker